src/test/scala/uk/gov/nationalarchives/tdr/keycloak/KeycloakUtilsTest.scala

```diff
@@ -21,13 +21,55 @@ class KeycloakUtilsTest extends AnyFlatSpec with Matchers with BeforeAndAfterEac
     mock.stop()
   }
 
-  "The verifyToken method " should "return a token for a valid token string " in {
-    val token = utils.verifyToken(mock.getAccessToken(aTokenConfig().build()))
-    assert(token.isDefined)
+  "The token method " should "return a bearer token for a valid token string " in {
+    val mockToken = mock.getAccessToken(aTokenConfig().build())
+    val token = utils.token(mockToken)
+    token.bearerAccessToken.getValue should equal(mockToken)
   }
 
-  "The verifyToken method " should "return a option empty for an invalid token string " in {
-    val token = utils.verifyToken("faketoken")
-    assert(token.isEmpty)
+  "The token method " should "return the correct user id for a valid token" in {
+    val userId = "1"
+    val mockToken = mock.getAccessToken(aTokenConfig().withClaim("user_id", userId).build())
+    val token = utils.token(mockToken)
+    token.userId should equal(Some(userId))
+  }
+
+  "The token method " should "return the correct transferring body for a valid token" in {
+    val body = "body"
+    val mockToken = mock.getAccessToken(aTokenConfig().withClaim("body", body).build())
+    val token = utils.token(mockToken)
+    token.transferringBody should equal(Some(body))
+  }
+
+  "The token method " should "return the correct roles for a valid token" in {
+    val role = "role_admin"
+    val mockToken = mock.getAccessToken(aTokenConfig().withResourceRole("tdr", role).build())
+    val token = utils.token(mockToken)
+    token.roles.size should be(1)
+    token.roles should contain(role)
+  }
+
+  "The verifyToken method " should "return a bearer token for an invalid token string " in {
```

Comment:
Should this read "The token method " ...

Reply:
It should indeed, thanks, I'll change it.